### ISSUE TITLE
spec(workspace/tune): re-establish Tune initial parameter spec (binary task) + inner_valid picker UI + Fit seed default 1120

### ISSUE THREAD

**Issue by nbx-liz** [nbx-liz/LizyStudio#459]
## Context

The user reports that the current Tune-tab initial settings have regressed from an earlier intended spec. Issue #458 already documents two narrower divergences (Re-tune Switch + BLUEPRINT default-range table). This issue **establishes the canonical spec for binary task** and tracks the implementation work to bring code + BLUEPRINT into agreement.

`regression` and `multiclass` defaults will be specified separately in a follow-up — this issue scope is **binary only**.

Issue #458 Finding B (BLUEPRINT default-range table is stale) is **superseded by this issue**. Issue #458 Finding A (Re-tune Switch) remains independent.

---

#### Smart Params

| Key | Mode | Initial value / Range / Choices |
|---|---|---|
| `auto_num_leaves` | Fixed | `True` |
| `num_leaves_ratio` | **Range** | low=0.4, high=1.0, log=false |
| `min_data_in_leaf_ratio` | **Range** | low=0.01, high=0.2, log=false |
| `min_data_in_bin_ratio` | **Range** | low=0.01, high=0.2, log=false |
| `feature_weights` | Fixed | disabled (`None`) |
| `balanced` | Fixed | `True` |

#### Model Params

| Key | Mode | Initial value / Range / Choices |
|---|---|---|
| `objective` | **Choice** | all options for task (binary: `[\"binary\", \"cross_entropy\", \"cross_entropy_lambda\"]`) |
| `metric` | **Choice** | binary: `[\"auc\", \"binary_logloss\", \"binary_error\", \"cross_entropy\", \"brier\"]` |
| `first_metric_only` | Fixed | `True` |
| `n_estimators` | **Range** | low=500, high=2000, log=false |
| `learning_rate` | **Range** | low=0.0001, high=0.01, log=true |
| `max_depth` | **Range** | low=3, high=9, log=false |
| `max_bin` | **Choice** | `[15, 63, 127, 255, 511]` |
| `feature_fraction` | **Range** | low=0.5, high=1.0, log=false |
| `bagging_fraction` | **Range** | low=0.5, high=1.0, log=false |
| `bagging_freq` | **Range** | low=1, high=10, log=false |
| `lambda_l1` | **Fixed** | `0` |
| `lambda_l2` | **Range** | low=0.000001, high=0.01, log=true |

#### Training

| Key | Mode | Initial value / Range / Choices |
|---|---|---|
| `seed` | Fixed | `1120` |
| `early_stopping.enabled` | Fixed | `True` |
| `early_stopping.rounds` | **Range** | low=50, high=200, log=false |
| `validation_ratio` | **Range** | low=0.1, high=0.3, log=false |
| `inner_valid` | Fixed | **CV-strategy-recommended default** (see below) |

### Tune Evaluation (`tuning.evaluation.metrics`) — binary

| Slot | Default |
|---|---|
| `metrics[0]` (Optimization Metric) | `auc` |
| `metrics[1..]` (Additional Metrics) | `[auc_pr, brier, logloss]` |

`direction` is auto-derived from `metric_direction[\"binary\"][\"auc\"] = \"maximize\"` (already implemented).

### `inner_valid` UI behavior — must follow CV strategy

The Tune-tab `inner_valid` row is currently rendered as a **free-text input** because `frontend/src/components/workspace/SearchSpaceRow.tsx` does not handle the `\"inner_valid_picker\"` value declared in `lizyml_ui_schema.py:574 special_search_space_fields[\"inner_valid\"]`.

Required behavior:

1. **UI:** Render a SegmentGroup of buttons (same visual pattern as `objective` and `metric` rows), with options filtered by the outer CV strategy via `cv-state.ts::filterInnerValidOptions(options, strategy)`:
   - Outer strategy ∈ {`kfold`, `stratified_kfold`}: options = `[\"holdout\"]`
   - Outer strategy ∈ {`group_kfold`, `stratified_group_kfold`, `blocked_group_kfold`}: options = `[\"holdout\", \"group_holdout\"]`
   - Outer strategy ∈ {`time_series`, `purged_time_series`, `group_time_series`}: options = `[\"holdout\", \"time_holdout\"]`
2. **Default value:** Use `cv-state.ts::recommendedInnerValid(strategy)` (which already exists for the Fit-tab inner_valid auto-reset). The catalog `default: \"holdout\"` should be replaced with a marker (e.g. `\"@cv_recommended\"`) or computed at render time.
3. **Auto-switch on outer-CV change:** Mirror the Fit-tab's `useConfigSync` reset effect — when the outer CV strategy changes, the Tune-tab `inner_valid` value updates to the new recommended default if the current value is no longer in the filtered options.

---

## Fit-tab Training Seed — default 1120

`lizyml.config.schema.TrainingConfig.seed = 42` (library SSOT). LizyStudio overrides this at the **default-config** layer so new workspaces start with `training.seed = 1120` (matching the Tune-tab catalog `seed` default already at 1120):

- File: `src/lizystudio/backends/lizyml/config_mixin.py::ConfigMixin.get_default_config()`
- Inject `\"training\": {\"seed\": 1120}` into the minimal config before `LizyMLConfig.model_validate()`.

This is a Studio-side UX default and does NOT change the lizyml schema. Users can still set any seed; only the initial fresh-config value changes.

---

### Spec docs (must land first)

1. **HISTORY.md**: file Proposal P-XXXX with this spec table verbatim, marking Decision row blank until implementation merges.
2. **BLUEPRINT.md S4.2.2** (L840-980): replace the L962-973 default-range table with the canonical spec above (Smart / Model / Training groups, all rows).

### Backend (lizyml_ui_schema.py)

3. Update `search_space_catalog` rows for: `num_leaves_ratio`, `min_data_in_leaf_ratio`, `min_data_in_bin_ratio` → add `default_mode: \"range\"`.
4. Update `objective`, `metric` → add `default_mode: \"choice\"` and per-task `default_choices` (binary list per spec; regression / multiclass rows stay current until follow-up).
5. Update `n_estimators`, `learning_rate`, `max_depth`, `max_bin`, `bagging_freq`, `early_stopping.rounds`, `lambda_l2` → adjust `default_range` / `default_choices` per spec.
6. `lambda_l1` → switch to `default_mode: \"fixed\"`, drop `default_range`, set `default: 0`.
7. `inner_valid` → keep `modes: [\"fixed\"]`, but mark the default as CV-recommended (one option: add a sentinel string `\"@cv_recommended\"` that the frontend resolves at render time, OR keep `default: \"holdout\"` but document that the frontend overrides it via `recommendedInnerValid`).
8. `config_mixin.py::get_default_config()` → inject `training.seed = 1120` override.

### Frontend (SearchSpaceRow.tsx)

9. Add `inner_valid_picker` branch in `SearchSpaceRow.tsx` mirroring the `objective` SegmentGroup branch:
   - `options` ← `filterInnerValidOptions(uiSchema.inner_valid_options, currentCvStrategy)`
   - `value` ← current `model.params.inner_valid` (or `recommendedInnerValid(strategy)` fallback)
   - `onChange` writes to `model.params.inner_valid` (same path as Fit tab's `training.early_stopping.inner_valid.method` — verify which field the Tune row maps to)
10. Wire `cvStrategy` prop down from `TuneTab.tsx` → `SearchSpaceTable.tsx` → `SearchSpaceRow.tsx` (read from `config.split.method`).
11. Add `useEffect` in `TuneTab.tsx` to auto-reset Tune-side `inner_valid` when outer CV strategy changes and current value is not in filtered options (mirror `ConfigForm.tsx:173`).

### Tune Evaluation defaults

12. `useJobResultData.ts` / `TuneEvaluationSection.tsx` — when `tuning.evaluation.metrics` is empty on first render, populate with task-specific defaults:
    - binary: `[\"auc\", \"auc_pr\", \"brier\", \"logloss\"]`
    - regression / multiclass: deferred to follow-up issue

### Tests

13. **Backend unit**: `tests/test_backends_lizyml.py` — assert `build_ui_schema()` rows match the canonical spec for each updated key.
14. **Backend unit**: `tests/test_backends_lizyml.py::test_get_default_config_binary_seed_1120`.
15. **Frontend Vitest**: `SearchSpaceRow.test.tsx` — render with `specialSearchSpaceFields[\"inner_valid\"] = \"inner_valid_picker\"` and verify SegmentGroup with strategy-filtered options renders.
16. **Frontend Vitest**: `TuneTab.test.tsx` — verify `inner_valid` auto-switches when CV strategy flips between (kfold) ↔ (time_series).
17. **Frontend Vitest**: `TuneEvaluationSection.test.tsx` — verify default optimization=auc + additional=[auc_pr, brier, logloss] for binary on fresh config.
18. **Contract** (optional but recommended): `tests/contract/test_blueprint_tune_spec_matches_ui_schema.py` — assert BLUEPRINT.md table matches `lizyml_ui_schema.py` row-by-row to prevent future drift.

---

## Out of scope

- `regression` and `multiclass` task defaults — follow-up issue once spec is provided.
- `num_leaves` Fixed-only mode — keep as-is unless explicitly requested.
- `verbose` Fixed -1 — not in user spec, keep unchanged.
- Re-tune `enabled` Switch (Issue #458 Finding A) — independent UX bug, ship separately.
- lizyml library-side `TrainingConfig.seed = 42` change — Studio-side override only.

---

### Spec
- [ ] HISTORY.md Proposal P-XXXX is filed and Decision row is updated when implementation merges.
- [ ] BLUEPRINT.md S4.2.2 default-range table is rewritten to match the canonical spec above (Smart / Model / Training groups).

### Search Space Catalog
- [ ] All 18 rows in the canonical spec match `lizyml_ui_schema.py::search_space_catalog`.
- [ ] `default_mode` is set to `\"range\"` / `\"choice\"` / `\"fixed\"` per the spec for each row.

### inner_valid picker
- [ ] Tune-tab `inner_valid` row renders as a SegmentGroup of buttons (not a free-text input).
- [ ] Options visible match `filterInnerValidOptions(strategy)` for the currently selected outer CV strategy.
- [ ] Initial value matches `recommendedInnerValid(strategy)`.
- [ ] Switching outer CV strategy auto-updates the Tune-tab `inner_valid` value.

### Tune Evaluation defaults (binary)
- [ ] Fresh binary tune config populates `tuning.evaluation.metrics = [\"auc\", \"auc_pr\", \"brier\", \"logloss\"]`.
- [ ] `optuna.params.direction = \"maximize\"` (auto-derived from `metric_direction[\"binary\"][\"auc\"]`).

### Fit-tab seed
- [ ] `GET /api/workspace/default-config?task=binary&target=...` returns `training.seed = 1120`.
- [ ] Existing tests for `training.seed = 42` (if any) are updated or removed.

### Process
- [ ] `pnpm test`, `pnpm test:e2e --grep workspace-tune`, `uv run pytest`, `pnpm check`, `pnpm build` all green.
- [ ] `frontend/src/api/generated/schema.d.ts` regenerated via `pnpm generate:api` if backend payload shape changes.

## Related

- Issue #458 (Re-tune Switch + BLUEPRINT stale) — Finding B superseded; Finding A independent
- BLUEPRINT.md S4.2.2 — Tune Tab UI
- `src/lizystudio/backends/lizyml_ui_schema.py` — search_space_catalog SSOT
- `src/lizystudio/backends/lizyml/config_mixin.py::get_default_config`
- `frontend/src/components/workspace/SearchSpaceRow.tsx` — needs inner_valid_picker branch
- `frontend/src/components/workspace/cv-state.ts` — `recommendedInnerValid`, `filterInnerValidOptions`
- `frontend/src/components/workspace/TuneEvaluationSection.tsx` — Optimization / Additional metric defaults
- HISTORY.md H-0061 / H-0068 — Re-tune lineage (referenced for context)

**Comment by nbx-liz** [comment]
## Update — multiclass + regression specs added; investigation findings

The user provided multiclass and regression specs and asked for related concerns to be investigated. Findings + extended canonical spec below.

---

## Investigation findings

### Finding 1: regression `objective` master list is contaminated with `cross_entropy` (BUG)

`src/lizystudio/backends/lizyml_ui_schema.py:36-43` declares:

```python
"regression": ["huber", "mse", "mae", "quantile", "mape", "cross_entropy"],
```

`cross_entropy` is a **binary classification** objective in LightGBM, not a valid regression objective. Picking it for a regression task makes LightGBM raise. This entry must be removed.

### Finding 2: multiclass optimization metric default = `auc` (preferred-metric path, technically valid but may surprise users)

`src/lizystudio/backends/lizyml_metrics.py:13-17`:

```python
_PREFERRED_METRIC: dict[str, str] = {
    "binary": "auc",
    "regression": "rmse",
    "multiclass": "auc",
}
```

`_TASK_METRICS["multiclass"] = {logloss, f1, accuracy, auc, auc_pr, brier}` so `auc` IS a registered multiclass metric (one-vs-rest macro). The user has decided to **keep `auc` as the multiclass optimization default** but pre-populate proper additional metrics (see spec below).

### Finding 3: Tune-tab additional metrics are never auto-populated

`frontend/src/components/workspace/TuneEvaluationSection.tsx:38-43` only fills the optimization slot from `metricOptions[0]`; the additional list stays empty until the user clicks chips. The user's spec for binary (`additional = [auc_pr, brier, logloss]`) is therefore a behavior change. The same pre-populate behavior applies to multiclass and regression.

---

## Canonical spec — regression task

### Search Space Catalog (regression)

| Group | Key | Mode | Initial value / Range / Choices |
|---|---|---|---|
| Smart | `auto_num_leaves` | Fixed | `True` |
| Smart | `num_leaves_ratio` | Range | low=0.4, high=1.0, log=false |
| Smart | `min_data_in_leaf_ratio` | Range | low=0.01, high=0.2, log=false |
| Smart | `min_data_in_bin_ratio` | Range | low=0.01, high=0.2, log=false |
| Smart | `feature_weights` | Fixed | disabled (`None`) |
| Smart | `balanced` | Fixed | `True` |
| Model | `objective` | **Choice** | default_choices = `[\"huber\", \"mse\", \"mae\"]` |
| Model | `metric` | **Choice** | default_choices = `[\"l1\", \"l2\", \"rmse\", \"huber\", \"rmsle\"]` |
| Model | `first_metric_only` | Fixed | `True` |
| Model | `n_estimators` | Range | low=500, high=2000, log=false |
| Model | `learning_rate` | Range | low=0.0001, high=0.01, log=true |
| Model | `max_depth` | Range | low=3, high=9, log=false |
| Model | `max_bin` | Choice | `[15, 63, 127, 255, 511]` |
| Model | `feature_fraction` | Range | low=0.5, high=1.0, log=false |
| Model | `bagging_fraction` | Range | low=0.5, high=1.0, log=false |
| Model | `bagging_freq` | Range | low=1, high=10, log=false |
| Model | `lambda_l1` | Fixed | `0` |
| Model | `lambda_l2` | Range | low=0.000001, high=0.01, log=true |
| Training | `seed` | Fixed | `1120` |
| Training | `early_stopping.enabled` | Fixed | `True` |
| Training | `early_stopping.rounds` | Range | low=50, high=200, log=false |
| Training | `validation_ratio` | Range | low=0.1, high=0.3, log=false |
| Training | `inner_valid` | Fixed | CV-strategy-recommended |

### Tune Evaluation (regression)

| Slot | Default |
|---|---|
| `metrics[0]` (Optimization) | `huber` |
| `metrics[1..]` (Additional) | `[mae, mape, rmsle, smape, wape]` |

`direction` auto-derived: `metric_direction[\"regression\"][\"huber\"] = \"minimize\"`.

### Bug fix — `option_sets.objective.regression`

| | Before | After |
|---|---|---|
| master option list | `[huber, mse, mae, quantile, mape, cross_entropy]` | `[huber, mse, mae, quantile, mape]` (drop `cross_entropy`) |
| Tune `default_choices` | (none — Fixed only) | `[huber, mse, mae]` |

`quantile` and `mape` remain available in the master list (Fit-tab dropdown) since they are valid LightGBM regression objectives, but the Tune-tab default search space includes only the 3 most common choices.

---

## Canonical spec — multiclass task

### Search Space Catalog (multiclass)

Identical to regression / binary for all rows except `objective` and `metric`:

| Group | Key | Mode | Initial value / Range / Choices |
|---|---|---|---|
| Model | `objective` | **Choice** | default_choices = `[\"multiclass\", \"softmax\", \"multiclassova\"]` |
| Model | `metric` | **Choice** | default_choices = `[\"multi_logloss\", \"multi_error\", \"auc_mu\", \"brier\"]` |

(All other rows: same as the canonical spec in the issue body.)

### Tune Evaluation (multiclass)

| Slot | Default |
|---|---|
| `metrics[0]` (Optimization) | `auc` |
| `metrics[1..]` (Additional) | `[logloss, f1, accuracy, auc_pr, brier]` |

`direction` auto-derived: `metric_direction[\"multiclass\"][\"auc\"] = \"maximize\"`.

`_PREFERRED_METRIC[\"multiclass\"]` stays at `auc` (no change to lizyml_metrics.py).

---

## Implementation deltas (additions to the original issue plan)

In addition to the binary plan already in the issue body:

1. **`option_sets.objective.regression`**: drop `cross_entropy` from the master list (bug fix). Verify no test or fixture depends on it.
2. **`search_space_catalog`** rows for `objective` and `metric`: per-task `default_choices` keyed by task — backend already supports this pattern (see `objective.default` for parameter_hints). Frontend `SearchSpaceRow` must read the task-keyed `default_choices` correctly.
3. **`TuneEvaluationSection.tsx` auto-populate**: when `tuning.evaluation.metrics` is empty on first render, write task-specific defaults:
   - binary: `[\"auc\", \"auc_pr\", \"brier\", \"logloss\"]`
   - regression: `[\"huber\", \"mae\", \"mape\", \"rmsle\", \"smape\", \"wape\"]`
   - multiclass: `[\"auc\", \"logloss\", \"f1\", \"accuracy\", \"auc_pr\", \"brier\"]`
   - Encoded as a constant map in the same module; safe to refactor later into `lizyml_ui_schema` if it grows.
4. **Backend regression test** for the `cross_entropy` removal: `tests/test_backends_lizyml.py::test_regression_objective_excludes_cross_entropy`.
5. **Backend test** for per-task `default_choices`: `tests/test_backends_lizyml.py::test_search_space_catalog_objective_default_choices_per_task` etc.

## Scope summary

This issue now covers all three tasks (binary / regression / multiclass) end-to-end. No follow-up issue is needed for additional task defaults.

**Comment by nbx-liz** [comment]
Resolved by:
- PR #468 (Wave 2.2) — canonical Range/Choice defaults aligned with LizyML binary spec + Fit `training.seed = 1120` default
- PR #470 (Wave 2.3) — inner_valid SegmentGroup picker UI + Tune Evaluation auto-populated default metrics

All acceptance criteria satisfied. See docs/handoff-2026-05-11-post-wave23.md for details.
